services/search/src/search/routes/search.py

```diff
@@ -0,0 +1,300 @@
+                    if length < 0:
+                        raise InvalidParameterError("Length must be positive")
+                    if length > MAX_ROWS:
+                        raise InvalidParameterError(f"Length must be less than or equal to {MAX_ROWS}")
+
+                with StepProfiler(method="search_endpoint", step="check authentication"):
+                    # if auth_check fails, it will raise an exception that will be caught below
+                    auth_check(
+                        dataset=dataset,
+                        external_auth_url=external_auth_url,
+                        request=request,
+                        hf_jwt_public_key=hf_jwt_public_key,
+                        hf_jwt_algorithm=hf_jwt_algorithm,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                    )
+
+                logging.info(f"/search {dataset=} {config=} {split=} {query=} {offset=} {length=}")
+
+                with StepProfiler(method="search_endpoint", step="validate indexing was done"):
+                    # no cache data is needed to download the index file
+                    # but will help to validate if indexing was done
+                    processing_steps = processing_graph.get_processing_step_by_job_type("split-duckdb-index")
+                    result = get_cache_entry_from_steps(
+                        processing_steps=[processing_steps],
+                        dataset=dataset,
+                        config=config,
+                        split=split,
+                        processing_graph=processing_graph,
+                        hf_endpoint=hf_endpoint,
+                        hf_token=hf_token,
+                        hf_timeout_seconds=hf_timeout_seconds,
+                        cache_max_days=cache_max_days,
+                    )
+
+                    content = result["content"]
+                    http_status = result["http_status"]
+                    error_code = result["error_code"]
+                    revision = result["dataset_git_revision"]
+                    if http_status != HTTPStatus.OK:
+                        return get_json_error_response(
+                            content=content,
+                            status_code=http_status,
+                            max_age=max_age_short,
+                            error_code=error_code,
+                            revision=revision,
+                        )
+
+                with StepProfiler(method="search_endpoint", step="download index file if missing"):
+                    index_folder = get_index_folder(duckdb_index_file_directory, dataset, config, split)
+                    repo_file_location = f"{config}/{split}/{DUCKDB_DEFAULT_INDEX_FILENAME}"
+                    index_file_location = f"{index_folder}/{repo_file_location}"
+                    index_path = Path(index_file_location)
+                    if not index_path.is_file():
```

Comment:
and what if the dataset has been updated but the old index is on the disk? we will search on an old version

Reply:
also: we don't delete the index files? It will be done by another job, right?

Reply:
Temporal index files will be deleted here https://github.com/huggingface/datasets-server/pull/1536 but as you said, this file also should be removed when there is a new version, maybe I can add that here? https://github.com/huggingface/datasets-server/blob/main/services/worker/src/worker/job_runners/split/duckdb_index.py#L73

Reply:
we should have a way to know if the current file is the correct version. Maybe use the revision in the path?

Reply:
OK, you take the revision in the name of the directory, so, we're good.